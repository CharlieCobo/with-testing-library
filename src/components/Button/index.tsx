import React from "react";
import { ButtonStyled } from "./styles";

interface ButtonProps {
  handleClick: (args: any) => any;
  label: string;
}

export const Button: React.FC<ButtonProps> = ({ handleClick, label }) => {
  return <ButtonStyled onClick={handleClick}>{label}</ButtonStyled>;
};
