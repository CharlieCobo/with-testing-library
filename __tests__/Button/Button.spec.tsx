import React from "react";
import { render, screen, act, fireEvent } from "@testing-library/react";
import { Button } from "../../src/components/Button";

const handleClick = jest.fn();

describe("<Button />", () => {
  test("should render the Button", () => {
    const { baseElement } = render(
      <Button handleClick={handleClick} label="Hello World" />
    );

    expect(baseElement).toMatchSnapshot();
  });

  test("should exec handleClick", () => {
    render(<Button handleClick={handleClick} label="Click Me!" />);

    const btn = screen.getByText("Click Me!");

    act(async () => await fireEvent.click(btn));

    expect(handleClick).toHaveBeenCalled();
  });
});
