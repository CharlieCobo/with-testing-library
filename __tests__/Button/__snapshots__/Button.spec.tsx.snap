// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the Button 1`] = `
.emotion-0 {
  color: red;
}

.emotion-0 {
  color: red;
}

<body>
  <div>
    <button
      class="emotion-0"
    >
      Hello World
    </button>
  </div>
</body>
`;
